### DISCUSSION TITLE
Cannot process CONNECT-method via server.

### DISCUSSION THREAD

**Discussion by ghost** [aio-libs/aiohttp#8118] [Q&A] [answered]
I am trying to make an aiohttp-server in order to handle CONNECT-requests:

```python
from aiohttp import web


async def process_request(request) -> web.Response:
    return web.Response()

app = web.Application()
import logging
logging.basicConfig(level=logging.DEBUG)
app.router.add_route('*', '/', process_request)
web.run_app(app, port=10001)
```

Starting server and trying to do some requests:
```
$ curl -v -X CONNECT http://0.0.0.0:10001/
*   Trying 0.0.0.0:10001...
* Connected to 0.0.0.0 (127.0.0.1) port 10001 (#0)
> CONNECT / HTTP/1.1
> Host: 0.0.0.0:10001
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Content-Type: text/plain; charset=utf-8
< Content-Length: 14
< Date: Wed, 31 Jan 2024 06:19:12 GMT
< Server: Python/3.11 aiohttp/3.9.3
< 
* Connection #0 to host 0.0.0.0 left intact
```

Server responded with 404 status.

Same path but with GET-method:
```
$ curl -v -X GET http://0.0.0.0:10001/
*   Trying 0.0.0.0:10001...
* Connected to 0.0.0.0 (127.0.0.1) port 10001 (#0)
> GET / HTTP/1.1
> Host: 0.0.0.0:10001
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Length: 0
< Content-Type: application/octet-stream
< Date: Wed, 31 Jan 2024 06:19:07 GMT
< Server: Python/3.11 aiohttp/3.9.3
< 
* Connection #0 to host 0.0.0.0 left intact
```

How I have 200 status code.

Server logs:
```
DEBUG:asyncio:Using selector: EpollSelector
======== Running on http://0.0.0.0:10001 ========
(Press CTRL+C to quit)
INFO:aiohttp.access:127.0.0.1 [31/Jan/2024:09:19:03 +0300] "GET / HTTP/1.1" 200 150 "-" "curl/7.81.0"
INFO:aiohttp.access:127.0.0.1 [31/Jan/2024:09:19:12 +0300] "CONNECT  HTTP/1.1" 404 173 "-" "curl/7.81.0"
```

What I am doing wrong? How to handle CONNECT-requests correctly?

**Comment by webknjaz** [comment, marked as answer]
aiohttp was never meant for making proxies, just regular web apps. I doubt the connect method reaches the HTTP handlers.

Perhaps you'd be able to intercept it via middlewares, I don't know. This would read the source code if you want to understand it better.

**Reply by ghost** [reply]
Thanks, you was right - connect method reaches the HTTP handlers.

The chain starts here where path (`/`) passed as `authority`. 
https://github.com/aio-libs/aiohttp/blob/f1e871b4e03e84fd6b60a37302e155114b9a30ed/aiohttp/_http_parser.pyx#L598

Futher processing leads to the fact that this simple `/` goes to `network` but `path` remains empty. Empty path will be resolved to propper handler anyway but not for `CONNECT` method.
